ledger-core/virtual/integration/constructor_test.go

```diff
@@ -288,3 +290,144 @@ func TestVirtual_Constructor_NoVFindCallRequestWhenMissing(t *testing.T) {
+		outgoingA     = server.RandomLocalWithPulse()
+		objectAGlobal = reference.NewSelf(outgoingA)
+
+		classB        = gen.UniqueReference()
+		objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+
+		outgoingCallRef = gen.UniqueReference()
+	)
+
+	// add ExecutionMocks to runnerMock
+	{
+		outgoingCall := executionevent.NewRPCBuilder(outgoingCallRef, objectAGlobal).CallConstructor(classB, "New", []byte("123"))
+		objectAResult := requestresult.New([]byte("finish A.New"), objectAGlobal)
+		objectAResult.SetActivate(reference.Global{}, classA, []byte("state A"))
+		objectAExecutionMock := runnerMock.AddExecutionMock(classA.String())
+		objectAExecutionMock.AddStart(
+			func(ctx execution.Context) {
+				t.Log("ExecutionStart [A.New]")
+				require.Equal(t, classA, ctx.Request.Callee)
+				require.Equal(t, outgoingA, ctx.Request.CallOutgoing)
+			},
+			&executionupdate.ContractExecutionStateUpdate{
+				Type:     executionupdate.OutgoingCall,
+				Error:    nil,
+				Outgoing: outgoingCall,
+			},
+		)
+		objectAExecutionMock.AddContinue(
+			func(result []byte) {
+				t.Log("ExecutionContinue [A.New]")
+				require.Equal(t, []byte("finish B.New"), result)
+			},
+			&executionupdate.ContractExecutionStateUpdate{
+				Type:   executionupdate.Done,
+				Result: objectAResult,
+			},
+		)
+
+		objectBResult := requestresult.New([]byte("finish B.New"), objectBGlobal)
+		objectBResult.SetActivate(reference.Global{}, classB, []byte("state B"))
+		runnerMock.AddExecutionMock(classB.String()).AddStart(
+			func(ctx execution.Context) {
+				t.Log("ExecutionStart [B.New]")
+				require.Equal(t, classB, ctx.Request.Callee)
+				require.Equal(t, objectAGlobal, ctx.Request.Caller)
+				require.Equal(t, []byte("123"), ctx.Request.Arguments)
+			},
+			&executionupdate.ContractExecutionStateUpdate{
+				Type:   executionupdate.Done,
+				Result: objectBResult,
+			},
+		)
+	}
+
+	// add checks to typedChecker
+	{
+		typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
+			assert.Equal(t, classB, request.Callee)
+			assert.Equal(t, objectAGlobal, request.Caller)
+			assert.Equal(t, []byte("123"), request.Arguments)
+			assert.Equal(t, payload.CTConstructor, request.CallType)
+			assert.Equal(t, uint32(1), request.CallSequence)
+			assert.Equal(t, outgoingCallRef, request.CallReason)
+			assert.Equal(t, callFlags, request.CallFlags)
+			assert.Equal(t, server.GetPulse().PulseNumber, request.CallOutgoing.Pulse())
+			return true // resend
+		})
+		typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+			assert.Equal(t, payload.CTConstructor, res.CallType)
+			assert.Equal(t, callFlags, res.CallFlags)
+
+			switch res.Callee {
+			case objectAGlobal:
+				require.Equal(t, []byte("finish A.New"), res.ReturnArguments)
+				require.Equal(t, server.GlobalCaller(), res.Caller)
+				require.Equal(t, outgoingA, res.CallOutgoing)
+			default:
+				require.Equal(t, []byte("finish B.New"), res.ReturnArguments)
```

Comment:
in this switch case you don't check Callee. I would recommend you to use `default:` for `t.FailNow("not expected Callee")`

Reply:
I don't know the real value of objectB Callee, because in stepExecuteOutgoing there is a generation of ID 
```
s.outgoing.CallOutgoing = gen.UniqueIDWithPulse(pulseNumber)
```
I can check it by pulseNumber, would it be better?

Reply:
you can save outgoing reference in VCallRequest check function a few lines above, save it outside function's scope.